fcrepo-client-impl/src/main/java/org/fcrepo/client/impl/FedoraResourceImpl.java

```diff
@@ -97,15 +97,118 @@ public FedoraResourceImpl(final FedoraRepository repository, final HttpHelper ht
+                throw new FedoraException("error copying resource " + uri + ": " + status.getStatusCode() + " " +
+                                     status.getReasonPhrase());
+            } else if (status.getStatusCode() == HttpStatus.SC_BAD_GATEWAY) {
+                // Destination URI isn't a valid resource path
+                LOGGER.error("error copying resource {}: {} {}", uri, status.getStatusCode(),
+                        status.getReasonPhrase());
+                throw new FedoraException("error copying resource " + uri + ": " + status.getStatusCode() + " " +
+                                     status.getReasonPhrase());
+            }
+        } catch (final FedoraException e) {
+            throw e;
+        } catch (final Exception e) {
+            LOGGER.error("could not encode URI parameter", e);
+            throw new FedoraException(e);
+        } finally {
+            copy.releaseConnection();
+        }
+    }
+
+    @Override
+    public void delete() throws FedoraException {
+        final HttpDelete delete = httpHelper.createDeleteMethod(path);
+
+        try {
+            final HttpResponse response = httpHelper.execute( delete );
+            final StatusLine status = response.getStatusLine();
+            final String uri = delete.getURI().toString();
+
+            if ( status.getStatusCode() == SC_NO_CONTENT) {
+                LOGGER.debug("triples updated successfully for resource {}", uri);
+            } else if ( status.getStatusCode() == SC_NOT_FOUND) {
+                LOGGER.error("resource {} does not exist, cannot update", uri);
+                throw new NotFoundException("resource " + uri + " does not exist, cannot update");
+            } else {
+                LOGGER.error("error updating resource {}: {} {}", uri, status.getStatusCode(),
+                             status.getReasonPhrase());
+                throw new FedoraException("error updating resource " + uri + ": " + status.getStatusCode() + " " +
+                                          status.getReasonPhrase());
+            }
+        } catch (final FedoraException e) {
+            throw e;
+        } catch (final Exception e) {
+            LOGGER.error("Error executing request", e);
+            throw new FedoraException(e);
+        } finally {
+            delete.releaseConnection();
+        }
     }
 
     @Override
-    public void delete() throws ReadOnlyException {
-        // TODO Auto-generated method stub
-        throw new NotImplemented("Method delete() is not implemented.");
+    public void forceDelete() throws FedoraException {
+            delete();
+            removeTombstone();
+    }
+
+    /**
+     * Remove tombstone (for the current path)
+     */
+    public void removeTombstone() throws FedoraException {
+        removeTombstone(path);
+    }
+
+
+    /**
+     * Remove tombstone located at given path
+     */
+    public void removeTombstone(final String path) throws FedoraException {
```

Comment:
Please add `removeTombstone` to the `FedoraResource` interface, or make the method private.